v2/pkg/genruntime/configmaps.go

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package genruntime
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/pkg/errors"
+
+	"github.com/Azure/azure-service-operator/v2/internal/set"
+)
+
+// ConfigMapReference is a reference to a Kubernetes configmap and key in the same namespace as
+// the resource it is on.
+// +kubebuilder:object:generate=true
+type ConfigMapReference struct {
+	// Name is the name of the Kubernetes configmap being referenced.
+	// The configmap must be in the same namespace as the resource
+	// +kubebuilder:validation:Required
+	Name string `json:"name"`
+
+	// Key is the key in the Kubernetes configmap being referenced
+	// +kubebuilder:validation:Required
+	Key string `json:"key"`
+}
+
+// Copy makes an independent copy of the ConfigMapReference
+func (c ConfigMapReference) Copy() ConfigMapReference {
+	return c
+}
+
+func (c ConfigMapReference) String() string {
+	return fmt.Sprintf("Name: %q, Key: %q", c.Name, c.Key)
+}
+
+// ToNamespacedRef creates a NamespacedSecretReference from this SecretReference in the given namespace
+func (c ConfigMapReference) ToNamespacedRef(namespace string) NamespacedConfigMapReference {
```

Comment:
Minor: I think we've been using the prefix `As` on these sorts of conversion methods elsewhere in ASO.

Reply:
`ResourceReference`, and `SecretReference` are already using the `To` prefix here, which was the pattern I was following for this type.

I went ahead and changed them all though as part of this PR to the `AsX` pattern.